src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -34,24 +34,32 @@ internal class TexlFunctionSet
 
         internal Dictionary<DPath, List<TexlFunction>>.KeyCollection Namespaces => _namespaces.Keys;
 
-        internal IEnumerable<TexlFunction> WithName(string name) => WithNameInternal(name);
+        internal IEnumerable<TexlFunction> WithName(string name) => _guard.VerifyNoWriters(this).WithNameInternal(name);
```

Comment:
These are key asserts - existing guards catch 2 writers. 
This check is for 1 reader during writers.